ESP32 compile error with arduino-esp32 v2.0.0-rc1 (error: 'rtc_gpio_desc' was not declared in this scope)

### Description

The OneWire library (latest version 2.3.5) fails to compile with arduino-esp32 v2.0.0-rc1 due to undeclared symbol 'rtc_gpio_desc'. This symbol apparently does not exist anymore under arduino-esp32 v2.0.0-rc1, and has been renamed to rtc_io_desc.


### Steps To Reproduce Problem

1. Install Arduino IDE
2. Install ESP32 support for Arduino, 2.0.0-rc1 (required for ESP32-S2 and ESP32-C3 support)
3. Install this library and some library requiring it
4. Compile any sketch that pulls OneWire (no need to flash anything, as this is a compile issue)


### Hardware & Software

Board: ESP32 (YUBOX)
Arduino IDE version: 1.8.15
Teensyduino version (if using Teensy)
Version info & package name (from Tools > Boards > Board Manager): esp32 by Espressif systems v2.0.0-rc1
Operating system & version: Fedora 34 x86_64
Any other software or hardware?


### Arduino Sketch

```cpp
#include <Arduino.h>
// Pulling header file is enough to trigger the compile error
#include <OneWire.h>

void setup() {
// No code required to trigger error
}

void loop() {
// No code required to trigger error
}
```


### Errors or Incorrect Output
```
In file included from /home/alex/Arduino/libraries/OneWire/OneWire.cpp:144:
/home/alex/Arduino/libraries/OneWire/util/OneWire_direct_gpio.h: In function 'void directModeInput(uint32_t)':
/home/alex/Arduino/libraries/OneWire/util/OneWire_direct_gpio.h:161:26: error: 'rtc_gpio_desc' was not declared in this scope
         uint32_t rtc_reg(rtc_gpio_desc[pin].reg);
                          ^~~~~~~~~~~~~
/home/alex/Arduino/libraries/OneWire/util/OneWire_direct_gpio.h:161:26: note: suggested alternative: 'rtc_io_desc'
         uint32_t rtc_reg(rtc_gpio_desc[pin].reg);
                          ^~~~~~~~~~~~~
                          rtc_io_desc
/home/alex/Arduino/libraries/OneWire/util/OneWire_direct_gpio.h: In function 'void directModeOutput(uint32_t)':
/home/alex/Arduino/libraries/OneWire/util/OneWire_direct_gpio.h:189:26: error: 'rtc_gpio_desc' was not declared in this scope
         uint32_t rtc_reg(rtc_gpio_desc[pin].reg);
                          ^~~~~~~~~~~~~
/home/alex/Arduino/libraries/OneWire/util/OneWire_direct_gpio.h:189:26: note: suggested alternative: 'rtc_io_desc'
         uint32_t rtc_reg(rtc_gpio_desc[pin].reg);
                          ^~~~~~~~~~~~~
                          rtc_io_desc
```

Today I installed a new version of ESP32, I think 2.0.0 came out and OneWire worked without a problem before that and now I have an error. Is there a way to fix this bug so I don't have to rework the entire code? I use DS18B20 on multiple pins, so I use OneWire.h together with another library <DallasTemperature.h> 

**I have ESP32-WROOM-32U**

`In file included from c:\Users\jsem\Documents\Arduino\libraries\OneWire\OneWire.cpp:144:
c:\Users\jsem\Documents\Arduino\libraries\OneWire\util/OneWire_direct_gpio.h: In function 'void directModeInput(uint32_t)':
c:\Users\jsem\Documents\Arduino\libraries\OneWire\util/OneWire_direct_gpio.h:161:26: error: 'rtc_gpio_desc' was not declared in this scope
         uint32_t rtc_reg(rtc_gpio_desc[pin].reg);
                          ^~~~~~~~~~~~~
c:\Users\jsem\Documents\Arduino\libraries\OneWire\util/OneWire_direct_gpio.h:161:26: note: suggested alternative: 'rtc_io_desc'
         uint32_t rtc_reg(rtc_gpio_desc[pin].reg);
                          ^~~~~~~~~~~~~
                          rtc_io_desc
c:\Users\jsem\Documents\Arduino\libraries\OneWire\util/OneWire_direct_gpio.h: In function 'void directModeOutput(uint32_t)':
c:\Users\jsem\Documents\Arduino\libraries\OneWire\util/OneWire_direct_gpio.h:189:26: error: 'rtc_gpio_desc' was not declared in this scope
         uint32_t rtc_reg(rtc_gpio_desc[pin].reg);
                          ^~~~~~~~~~~~~
c:\Users\jsem\Documents\Arduino\libraries\OneWire\util/OneWire_direct_gpio.h:189:26: note: suggested alternative: 'rtc_io_desc'
         uint32_t rtc_reg(rtc_gpio_desc[pin].reg);
                          ^~~~~~~~~~~~~
                          rtc_io_desc
Compilation error: Error: 2 UNKNOWN: exit status 1
`

--- a/util/OneWire_direct_gpio.h
+++ b/util/OneWire_direct_gpio.h
@@ -158,6 +158,18 @@ void directModeInput(IO_REG_TYPE pin)
 {
     if ( digitalPinIsValid(pin) )
     {
+        #if defined(ESP_ARDUINO_VERSION)
+        #if ESP_ARDUINO_VERSION >= ESP_ARDUINO_VERSION_VAL(2, 0, 0)
+        int pin_io = rtc_io_number_get((gpio_num_t)pin);
+        uint32_t rtc_reg(rtc_io_desc[pin_io].reg);
+
+        if ( rtc_reg ) // RTC pins PULL settings
+        {
+            ESP_REG(rtc_reg) = ESP_REG(rtc_reg) & ~(rtc_io_desc[pin_io].mux);
+            ESP_REG(rtc_reg) = ESP_REG(rtc_reg) & ~(rtc_io_desc[pin_io].pullup | rtc_io_desc[pin_io].pulldown);
+        }
+        #endif
+        #else
         uint32_t rtc_reg(rtc_gpio_desc[pin].reg);
 
         if ( rtc_reg ) // RTC pins PULL settings
@@ -165,6 +177,7 @@ void directModeInput(IO_REG_TYPE pin)
             ESP_REG(rtc_reg) = ESP_REG(rtc_reg) & ~(rtc_gpio_desc[pin].mux);
             ESP_REG(rtc_reg) = ESP_REG(rtc_reg) & ~(rtc_gpio_desc[pin].pullup | rtc_gpio_desc[pin].pulldown);
         }
+        #endif
 
         if ( pin < 32 )
             GPIO.enable_w1tc = ((uint32_t)1 << pin);
@@ -186,6 +199,18 @@ void directModeOutput(IO_REG_TYPE pin)
 {
     if ( digitalPinIsValid(pin) && pin <= 33 ) // pins above 33 can be only inputs
     {
+        #if defined(ESP_ARDUINO_VERSION)
+        #if ESP_ARDUINO_VERSION >= ESP_ARDUINO_VERSION_VAL(2, 0, 0)
+        int pin_io = rtc_io_number_get((gpio_num_t)pin);
+        uint32_t rtc_reg(rtc_io_desc[pin_io].reg);
+
+        if ( rtc_reg ) // RTC pins PULL settings
+        {
+            ESP_REG(rtc_reg) = ESP_REG(rtc_reg) & ~(rtc_io_desc[pin_io].mux);
+            ESP_REG(rtc_reg) = ESP_REG(rtc_reg) & ~(rtc_io_desc[pin_io].pullup | rtc_io_desc[pin_io].pulldown);
+        }
+        #endif
+        #else
         uint32_t rtc_reg(rtc_gpio_desc[pin].reg);
 
         if ( rtc_reg ) // RTC pins PULL settings
@@ -193,6 +218,7 @@ void directModeOutput(IO_REG_TYPE pin)
             ESP_REG(rtc_reg) = ESP_REG(rtc_reg) & ~(rtc_gpio_desc[pin].mux);
             ESP_REG(rtc_reg) = ESP_REG(rtc_reg) & ~(rtc_gpio_desc[pin].pullup | rtc_gpio_desc[pin].pulldown);
         }
+        #endif
 
         if ( pin < 32 )
             GPIO.enable_w1ts = ((uint32_t)1 << pin);

